pkg/api/apitest/apitest.go

```diff
@@ -0,0 +1,162 @@
+package apitest
+
+import (
+	crand "crypto/rand"
+	"encoding/json"
+	"math/rand"
+
+	"github.com/blang/semver"
+	"github.com/oklog/ulid"
+	"github.com/rvolosatovs/turtlitto/pkg/api"
+)
+
+func RandomMessage() api.Message {
+	var msg = api.Message{}
+	msg.Type = *RandomMessageType()
+	msg.MessageID = *RandomULID()
+
+	var pld interface{}
+	switch msg.Type {
+	case api.MessageTypeHandshake:
+		pld = *RandomHandshake()
+	case api.MessageTypeState:
+		pld = *RandomState()
+	case api.MessageTypePing:
+		pld = nil
+	default:
+		panic("unmatched Message type")
+	}
+
+	b, err := json.Marshal(pld)
+	if err != nil {
+		panic("could not marshall payload ")
+	}
+
+	msg.Payload = b
+	return msg
+}
+
+func RandomMessageType() *api.MessageType {
+	vals := []api.MessageType{api.MessageTypeState, api.MessageTypePing, api.MessageTypeHandshake}
+	return &vals[rand.Intn(len(vals))]
+}
+
+func RandomHandshake() *api.Handshake {
+	//TODO: Proper way of creating handshakes
+	vers := string(rand.Intn(10)) + "." + string(rand.Intn(10)) + "." + string(rand.Intn(10))
```

Comment:
is random version a good idea?

Reply:
TRC revisions are a number, so I suggest adhering to something like [TenVer](http://tenver.drone-ver.org/), or just return a random number between 0..99999

Reply:
It's not extensible and is not what we defined in the spec